### PR TITLE
Selenium Testing Improvements

### PR DESCRIPTION
**Pull request by jmchilton**
- Initial collection builders tests to start working on test cases for #3696.
- Improved exception messages for more tests to help with tracking down problems with remaining transient test failures.

### CODE REVIEW

File: test/galaxy_selenium/has_driver.py [hunk 1]
```diff
@@ -14,6 +15,7 @@
 
 
 class HasDriver:
+    TimeoutException = TimeoutException
```
Comment (nsoranzo) [inline test/galaxy_selenium/has_driver.py:18]: Is that needed?

Comment (nsoranzo) [reply test/galaxy_selenium/has_driver.py:18]: I guess because it's used as `self.TimeoutException` in `NavigatesGalaxy` subclass, sorry for the noise

Comment (jmchilton) [reply test/galaxy_selenium/has_driver.py:18]: I have avoided selenium imports in ``test/galaxy_selenium/navigates_galaxy.py`` and all of the ``test/selenium_tests`` files - this was required to to continue that. I didn't necessarily set out to do be so encapsulated - but it is a nice property to have now.

If you insist I will drop this and add a direct import of this class from ``navigates_galaxy.py``.

Comment (nsoranzo) [reply test/galaxy_selenium/has_driver.py:18]: That's fine.

**Pull request merged**